tofhir-engine/src/test/scala/io/tofhir/test/engine/execution/FileStreamInputArchiverTest.scala

```diff
@@ -0,0 +1,367 @@
+package io.tofhir.test.engine.execution
+
+import io.tofhir.engine.config.ToFhirConfig
+import io.tofhir.engine.execution.FileStreamInputArchiver.getOffsetKey
+import io.tofhir.engine.execution.{FileStreamInputArchiver, RunningJobRegistry}
+import io.tofhir.engine.model._
+import io.tofhir.engine.util.FileUtils
+import org.mockito.MockitoSugar._
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.io.{File, PrintWriter}
+import scala.reflect.runtime.universe._
+
+class FileStreamInputArchiverTest extends AnyFlatSpec with Matchers {
+
+  val runningJobRegistryMock = mock[RunningJobRegistry]
+  val fileStreamInputArchiver = new FileStreamInputArchiver(runningJobRegistryMock)
+
+  "FileStreamInputArchiver" should "apply archiving for a streaming job" in {
+
+    // Create a mock execution
+    val mockTaskExecution = mock[FhirMappingJobExecution]
```

Comment:
Can you convert this to real object instead of mocking, as you are duplicating the logic with directories, which is error-prone and makes maintenance harder.